### ISSUE TITLE
Scaling Resolution Issues and Pixel Distortion

### ISSUE THREAD

**Issue by g-brrzzn** [g-brrzzn/shootemup-Pygame#2]
**Description:**
Currently, the game is being rendered directly at the window's resolution, which is causing scaling issues and pixel distortion, especially in fullscreen mode or at higher resolutions.

**Steps to Reproduce:**
Run the game in fullscreen mode or at a resolution different from 1280x720.
Observe the pixel distortion and stretching effect on the game's graphical elements.

**Expected Behavior:**
The game screen should be rendered at a fixed resolution (e.g., 1280x720) and then scaled to the window's resolution, maintaining the pixel aspect ratio.

**Proposed Solution:**
Implement an intermediate rendering buffer (Surface) where all the game drawing logic is executed, and then scale this Surface to the current window resolution using pygame.transform.scale().

**Comment by abhipxx** [comment]
Hey! Can I try contributing to this?

**Comment by g-brrzzn** [comment]
Absolutely, feel free to give it a try! If you need any clarification, let me know and I’ll be happy to help.
